engine/src/flutter/shell/platform/linux/fl_renderer.cc

```diff
@@ -564,6 +641,41 @@ gboolean fl_renderer_present_layers(FlRenderer* self,
   return TRUE;
 }
 
+struct _FlRendererPresentLayersData {
+  FlRenderer* self;
+  FlutterViewId view_id;
+  const FlutterLayer** layers;
+  size_t layers_count;
+  gboolean* res;
+};
+
+static void fl_renderer_present_layers_trampoline(gpointer trampoline_data) {
+  _FlRendererPresentLayersData* data =
+      reinterpret_cast<_FlRendererPresentLayersData*>(trampoline_data);
+  fl_renderer_make_current(data->self);
+  *data->res = fl_renderer_present_layers_on_main_thread(
+      data->self, data->view_id, data->layers, data->layers_count);
+  fl_renderer_clear_current(data->self);
+}
+
+gboolean fl_renderer_present_layers(FlRenderer* self,
+                                    FlutterViewId view_id,
+                                    const FlutterLayer** layers,
+                                    size_t layers_count) {
+  gboolean res = FALSE;
+  _FlRendererPresentLayersData data = {
+      .self = self,
+      .view_id = view_id,
+      .layers = layers,
+      .layers_count = layers_count,
+      .res = &res,
+  };
+  fl_renderer_clear_current(self);
+  fl_renderer_call_blocking(self, fl_renderer_present_layers_trampoline, &data);
```

Comment:
I think this can be removed and [`g_idle_add_full`](https://docs.gtk.org/glib/func.idle_add_full.html) used (or [`g_source_attach`](https://docs.gtk.org/glib/method.Source.attach.html) if need more control).

Reply:
We can't use `g_idle_add` here because during resizing, the `FlTaskRunner` works in a special mode where only tasks posted form Flutter are executed and `GMainContext` is not being pumped. Hence modification to `FlTaskRunner` to schedule arbitrary callbacks alongside Flutter tasks.